api/src/main/java/com/nicknathanjustin/streamercontracts/contracts/ContractServiceImpl.java

```diff
@@ -61,12 +65,26 @@ public ContractModel createContract(@NonNull final UserModel proposer, @NonNull
 
     @Override
     public void setContractState(@NonNull final ContractModel contractModel, final ContractState newContractState) {
-        // TODO: loop through donations for the contract and either void or capture payPalPayments when the contract
-        // transitions to a "payable" state
         contractModel.setContractState(newContractState);
         contractModelRepository.save(contractModel);
     }
 
+    @Override
+    public void settlePayments(@NonNull final ContractModel contractModel) {
+        final List payStreamerStates = ImmutableList.of(ContractState.COMPLETED, ContractState.DISPUTED);
+        final List payDonorStates = ImmutableList.of(ContractState.DECLINED, ContractState.FAILED, ContractState.EXPIRED);
```

Comment:
I prefer Immutable List since it conveys the intent that the list should not have items added or removed.

What warnings does java say? My intellij doesnt have any.

Reply:
Maybe my knowledge of Java warnings is out of date but I thought Java complained if explicit types weren't used